protected/humhub/modules/file/models/File.php

```diff
@@ -112,12 +139,37 @@ public function behaviors()
     }
 
     /**
-     * Gets query for [[FileHistory]].
+     * @return array
+     * @noinspection PhpMissingReturnTypeInspection
+     * @noinspection ReturnTypeCanBeDeclaredInspection
+     */
+    public function transactions()
+    {
+        return [
+            'default' => self::OP_INSERT + self::OP_DELETE,
+        ];
+    }
+
+    public function __get($name)
+    {
+        $value = parent::__get($name);
+
+        if ($name === 'metadata' && !$value instanceof Metadata) {
```

Comment:
What about using this here?
```php
if ($name === 'metadata') {
   return $this->getMetadata();
}
```

Reply:
Yes, that works, too!

Your solution is certainly easier to understand. My solution was maybe minimally more efficient on successive calls ...